cmd/vsphere-xcopy-volume-populator/README.md

```diff
@@ -162,6 +162,253 @@ to have a secret with those required fields
+2. **Manual SSH key deployment** - the system will provide instructions in logs if keys need to be installed
+3. Once SSH and keys are configured, the system sets up secure command restrictions automatically
+
+**Important**: SSH service enablement and initial key deployment must be done manually. The system will detect missing keys and provide step-by-step instructions in the logs.
+
+## 7. Manual SSH Key Installation
+
+You can prepare your hosts for the SSH method by manually adding the restricted SSH key to your ESXi hosts using the following steps. 
+
+**Note**: A simpler approach is to let one migration fail and follow the instructions in the populator pod logs - they will have the exact key and datastore path filled in for you.
+
+If you want to configure SSH keys prior to the first migration:
+
+### Step 1: Extract the Public Key
+
+First, get the public key from the auto-generated secret:
+
+```bash
+# List SSH key secrets to find the right one
+oc get secrets -l app.kubernetes.io/component=ssh-keys -n openshift-mtv
+
+# Extract the public key (replace with your actual secret name)
+oc get secret offload-ssh-keys-https---vcenter-example-com-public \
+  -o jsonpath='{.data.public-key}' -n openshift-mtv | base64 -d > esxi_public_key.pub
+
+# View the public key
+cat esxi_public_key.pub
+```
+
+### Step 2: Prepare the Restricted Key Entry
+
+The system requires command restrictions for security. Create the restricted key entry:
+
+```bash
+# The public key needs to be prefixed with command restrictions
+# The script path will be: /vmfs/volumes/{datastore-name}/secure-vmkfstools-wrapper.py
+# Replace {datastore-name} with your actual datastore name
+echo 'command="python /vmfs/volumes/datastore1/secure-vmkfstools-wrapper.py",no-port-forwarding,no-agent-forwarding,no-X11-forwarding '$(cat esxi_public_key.pub) > restricted_key.pub
+
+# View the final restricted key
+cat restricted_key.pub
+```
+
+### Step 3: Install the Key on ESXi Host
+
+Connect to each ESXi host and install the key:
+
+```bash
+# SSH to the ESXi host as root
+ssh root@esxi-host-ip
+
+# Create the SSH keys directory if it doesn't exist
+mkdir -p /etc/ssh/keys-root
+
+# Add the restricted public key to authorized_keys
+# Copy the content from restricted_key.pub and paste it into the file
+vi /etc/ssh/keys-root/authorized_keys
+
+# Set proper permissions
+chmod 600 /etc/ssh/keys-root/authorized_keys
+chmod 700 /etc/ssh/keys-root
+
+# Restart SSH service to apply changes
+/etc/init.d/SSH restart
+```
+
+### Step 4: Alternative - One-Command Installation
+
+If you have network access from your local machine to the ESXi host:
+
+```bash
+# Copy the restricted key directly (replace with your ESXi IP)
+cat restricted_key.pub | ssh root@esxi-host-ip \
+  'mkdir -p /etc/ssh/keys-root && cat >> /etc/ssh/keys-root/authorized_keys && chmod 600 /etc/ssh/keys-root/authorized_keys && chmod 700 /etc/ssh/keys-root && /etc/init.d/SSH restart'
```

Comment:
the directory can exist and this whole line will fail. 
also, if the dir exists, and the ssh daemon is already configured to using that authorized keys file then there is no need to restart any daemon, just amend the file.